src/main/java/io/aiven/kafka/connect/transforms/FilterByFieldValue.java

```diff
@@ -0,0 +1,196 @@
+        }
+        Optional<String> text = Optional.empty();
+        if (isSupportedType(fieldValue)) {
+            text = Optional.of(fieldValue.toString());
+        }
+        return text;
+    }
+
+    private boolean isSupportedType(final Object fieldValue) {
+        final Set<Class<?>> supportedTypes = new HashSet<>(
+            Arrays.asList(
+                String.class, Long.class, Integer.class, Short.class,
+                Double.class, Float.class, Boolean.class
+            )
+        );
+
+        return supportedTypes.contains(fieldValue.getClass());
+    }
+
+    @Override
+    public ConfigDef config() {
+        return new ConfigDef()
+            .define("field.name",
+                ConfigDef.Type.STRING,
+                null,
+                ConfigDef.Importance.HIGH,
+                "The field name to filter by."
+                    + "Schema-based records (Avro), schemaless (e.g. JSON), and raw values are supported."
+                    + "If empty, the whole key/value record will be filtered.")
+            .define("field.value",
+                ConfigDef.Type.STRING,
+                null,
+                ConfigDef.Importance.HIGH,
+                "Expected value to match. Either define this, or a regex pattern")
+            .define("field.value.pattern",
+                ConfigDef.Type.STRING,
+                null,
+                ConfigDef.Importance.HIGH,
+                "The pattern to match. Either define this, or an expected value")
+            .define("field.value.matches",
+                ConfigDef.Type.BOOLEAN,
+                true,
+                ConfigDef.Importance.MEDIUM,
+                "The filter mode, 'true' for matching or 'false' for non-matching");
+    }
+
+    @Override
+    public void close() {
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        final AbstractConfig config = new AbstractConfig(config(), configs);
+        this.fieldName = config.getString("field.name");
+        this.fieldExpectedValue = Optional.ofNullable(config.getString("field.value"));
+        this.fieldValuePattern = Optional.ofNullable(config.getString("field.value.pattern"));
+        final boolean expectedValuePresent = fieldExpectedValue.map(s -> !s.isEmpty()).orElse(false);
+        final boolean regexPatternPresent = fieldValuePattern.map(s -> !s.isEmpty()).orElse(false);
+        if ((expectedValuePresent && regexPatternPresent)
+            || (!expectedValuePresent && !regexPatternPresent)) {
+            throw new ConfigException(
+                "Either field.value or field.value.pattern have to be set to apply filter transform");
+        }
+        final Predicate<Optional<String>> matchCondition = fieldValue -> fieldValue
+            .filter(value -> expectedValuePresent
+                ? fieldExpectedValue.get().equals(value)
+                : value.matches(fieldValuePattern.get()))
+            .isPresent();
```

Comment:
We should put the `fieldExpectedValue::get` and `fieldValuePattern::get` calls and if/else logic outside the predicate so that it's only evaluated once (small optimizations like this are worthwhile because this predicate will be applied to every record in a connector's pipeline).

Reply:
Now the logic is only evaluated once during the configuration rather than for each record.